python/cudf/cudf/core/dtypes.py

```diff
@@ -39,6 +39,26 @@
     from cudf.core.buffer import Buffer
 
 
+DTYPE_ENUM_MAP = {
+    PANDAS_NUMPY_DTYPE: 1,
+    pd.core.dtypes.dtypes.ArrowDtype: 2,
+    pd.core.dtypes.dtypes.ExtensionDtype: 3,
+}
+
+
+def get_dtype_enum(dtype: Dtype) -> int:
```

Comment:
I would be nice to use an `enum.Enum` to represent this so at least in the code we could check e.g. `PandasTypeEnum.ARROW` or something